lci_strategic_plugin/include/lci_strategic_plugin/lci_strategic_plugin.h

```diff
@@ -513,6 +513,22 @@ class LCIStrategicPlugin
                                                                      bool shortest_path_only = false,
                                                                      bool bounds_inclusive = true) const;
   
+  /**
+   * \brief Get the final entry time from either vehicle's own internal ET calculation (TSMO UC2) or from carma-street (TSMO UC3)
+   *        This function also applies necessary green_buffer to adjust the ET. 
+   * \param current_state Current state of the vehicle
+   * \param earliest_entry_time Earliest entry time calculated by the vehicle
+   * \param traffic_light traffic signal the vehicle is using
+   * \param is_entry_time_within_green_or_tdb this function sets it to true if ET is in green or TDB (related to UC3, always set to true in UC2)
+   * \param in_tdb this function sets it to true if ET is in TDB (not modified if in UC2)
```

Comment:
Not a huge fan of return via parameter reference, as a general rule. Especially since "in_tbd" reads like an input variable name because of the "in" part.

TBH, I'd recommend either defining a struct when multiple return values are necessary or simply using [`std::tuple<...>` ](https://en.cppreference.com/w/cpp/utility/tuple). std::tuple is probably more readable if we're able to use the structured binding syntax mentioned in that documentation link.

Reply:
I went ahead and made them to tuple